// @mistralai/mistralai/models/components/batchjobin.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BatchJobIn$ = exports.BatchJobIn$outboundSchema = exports.BatchJobIn$inboundSchema = void 0;
exports.batchJobInToJSON = batchJobInToJSON;
exports.batchJobInFromJSON = batchJobInFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const apiendpoint_js_1 = require("./apiendpoint.js");
/** @internal */
exports.BatchJobIn$inboundSchema = z.object({
    input_files: z.array(z.string()),
    endpoint: apiendpoint_js_1.ApiEndpoint$inboundSchema,
    model: z.string(),
    metadata: z.nullable(z.record(z.string())).optional(),
    timeout_hours: z.number().int().default(24),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "input_files": "inputFiles",
        "timeout_hours": "timeoutHours",
    });
});
/** @internal */
exports.BatchJobIn$outboundSchema = z.object({
    inputFiles: z.array(z.string()),
    endpoint: apiendpoint_js_1.ApiEndpoint$outboundSchema,
    model: z.string(),
    metadata: z.nullable(z.record(z.string())).optional(),
    timeoutHours: z.number().int().default(24),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        inputFiles: "input_files",
        timeoutHours: "timeout_hours",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var BatchJobIn$;
(function (BatchJobIn$) {
    /** @deprecated use `BatchJobIn$inboundSchema` instead. */
    BatchJobIn$.inboundSchema = exports.BatchJobIn$inboundSchema;
    /** @deprecated use `BatchJobIn$outboundSchema` instead. */
    BatchJobIn$.outboundSchema = exports.BatchJobIn$outboundSchema;
})(BatchJobIn$ || (exports.BatchJobIn$ = BatchJobIn$ = {}));
function batchJobInToJSON(batchJobIn) {
    return JSON.stringify(exports.BatchJobIn$outboundSchema.parse(batchJobIn));
}
function batchJobInFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.BatchJobIn$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'BatchJobIn' from JSON`);
}
