// @mistralai/mistralai/models/components/classificationresponse.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ClassificationResponse$ = exports.ClassificationResponse$outboundSchema = exports.ClassificationResponse$inboundSchema = void 0;
exports.classificationResponseToJSON = classificationResponseToJSON;
exports.classificationResponseFromJSON = classificationResponseFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const classificationobject_js_1 = require("./classificationobject.js");
/** @internal */
exports.ClassificationResponse$inboundSchema = z.object({
    id: z.string().optional(),
    model: z.string().optional(),
    results: z.array(classificationobject_js_1.ClassificationObject$inboundSchema).optional(),
});
/** @internal */
exports.ClassificationResponse$outboundSchema = z.object({
    id: z.string().optional(),
    model: z.string().optional(),
    results: z.array(classificationobject_js_1.ClassificationObject$outboundSchema).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var ClassificationResponse$;
(function (ClassificationResponse$) {
    /** @deprecated use `ClassificationResponse$inboundSchema` instead. */
    ClassificationResponse$.inboundSchema = exports.ClassificationResponse$inboundSchema;
    /** @deprecated use `ClassificationResponse$outboundSchema` instead. */
    ClassificationResponse$.outboundSchema = exports.ClassificationResponse$outboundSchema;
})(ClassificationResponse$ || (exports.ClassificationResponse$ = ClassificationResponse$ = {}));
function classificationResponseToJSON(classificationResponse) {
    return JSON.stringify(exports.ClassificationResponse$outboundSchema.parse(classificationResponse));
}
function classificationResponseFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.ClassificationResponse$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'ClassificationResponse' from JSON`);
}
