// @mistralai/mistralai/models/components/detailedjobout.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DetailedJobOut$ = exports.DetailedJobOut$outboundSchema = exports.DetailedJobOut$inboundSchema = exports.DetailedJobOutRepositories$ = exports.DetailedJobOutRepositories$outboundSchema = exports.DetailedJobOutRepositories$inboundSchema = exports.DetailedJobOutIntegrations$ = exports.DetailedJobOutIntegrations$outboundSchema = exports.DetailedJobOutIntegrations$inboundSchema = exports.DetailedJobOutObject$ = exports.DetailedJobOutObject$outboundSchema = exports.DetailedJobOutObject$inboundSchema = exports.DetailedJobOutStatus$ = exports.DetailedJobOutStatus$outboundSchema = exports.DetailedJobOutStatus$inboundSchema = exports.DetailedJobOutObject = exports.DetailedJobOutStatus = void 0;
exports.detailedJobOutIntegrationsToJSON = detailedJobOutIntegrationsToJSON;
exports.detailedJobOutIntegrationsFromJSON = detailedJobOutIntegrationsFromJSON;
exports.detailedJobOutRepositoriesToJSON = detailedJobOutRepositoriesToJSON;
exports.detailedJobOutRepositoriesFromJSON = detailedJobOutRepositoriesFromJSON;
exports.detailedJobOutToJSON = detailedJobOutToJSON;
exports.detailedJobOutFromJSON = detailedJobOutFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const checkpointout_js_1 = require("./checkpointout.js");
const eventout_js_1 = require("./eventout.js");
const githubrepositoryout_js_1 = require("./githubrepositoryout.js");
const jobmetadataout_js_1 = require("./jobmetadataout.js");
const trainingparameters_js_1 = require("./trainingparameters.js");
const wandbintegrationout_js_1 = require("./wandbintegrationout.js");
exports.DetailedJobOutStatus = {
    Queued: "QUEUED",
    Started: "STARTED",
    Validating: "VALIDATING",
    Validated: "VALIDATED",
    Running: "RUNNING",
    FailedValidation: "FAILED_VALIDATION",
    Failed: "FAILED",
    Success: "SUCCESS",
    Cancelled: "CANCELLED",
    CancellationRequested: "CANCELLATION_REQUESTED",
};
exports.DetailedJobOutObject = {
    Job: "job",
};
/** @internal */
exports.DetailedJobOutStatus$inboundSchema = z.nativeEnum(exports.DetailedJobOutStatus);
/** @internal */
exports.DetailedJobOutStatus$outboundSchema = exports.DetailedJobOutStatus$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var DetailedJobOutStatus$;
(function (DetailedJobOutStatus$) {
    /** @deprecated use `DetailedJobOutStatus$inboundSchema` instead. */
    DetailedJobOutStatus$.inboundSchema = exports.DetailedJobOutStatus$inboundSchema;
    /** @deprecated use `DetailedJobOutStatus$outboundSchema` instead. */
    DetailedJobOutStatus$.outboundSchema = exports.DetailedJobOutStatus$outboundSchema;
})(DetailedJobOutStatus$ || (exports.DetailedJobOutStatus$ = DetailedJobOutStatus$ = {}));
/** @internal */
exports.DetailedJobOutObject$inboundSchema = z.nativeEnum(exports.DetailedJobOutObject);
/** @internal */
exports.DetailedJobOutObject$outboundSchema = exports.DetailedJobOutObject$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var DetailedJobOutObject$;
(function (DetailedJobOutObject$) {
    /** @deprecated use `DetailedJobOutObject$inboundSchema` instead. */
    DetailedJobOutObject$.inboundSchema = exports.DetailedJobOutObject$inboundSchema;
    /** @deprecated use `DetailedJobOutObject$outboundSchema` instead. */
    DetailedJobOutObject$.outboundSchema = exports.DetailedJobOutObject$outboundSchema;
})(DetailedJobOutObject$ || (exports.DetailedJobOutObject$ = DetailedJobOutObject$ = {}));
/** @internal */
exports.DetailedJobOutIntegrations$inboundSchema = wandbintegrationout_js_1.WandbIntegrationOut$inboundSchema;
/** @internal */
exports.DetailedJobOutIntegrations$outboundSchema = wandbintegrationout_js_1.WandbIntegrationOut$outboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var DetailedJobOutIntegrations$;
(function (DetailedJobOutIntegrations$) {
    /** @deprecated use `DetailedJobOutIntegrations$inboundSchema` instead. */
    DetailedJobOutIntegrations$.inboundSchema = exports.DetailedJobOutIntegrations$inboundSchema;
    /** @deprecated use `DetailedJobOutIntegrations$outboundSchema` instead. */
    DetailedJobOutIntegrations$.outboundSchema = exports.DetailedJobOutIntegrations$outboundSchema;
})(DetailedJobOutIntegrations$ || (exports.DetailedJobOutIntegrations$ = DetailedJobOutIntegrations$ = {}));
function detailedJobOutIntegrationsToJSON(detailedJobOutIntegrations) {
    return JSON.stringify(exports.DetailedJobOutIntegrations$outboundSchema.parse(detailedJobOutIntegrations));
}
function detailedJobOutIntegrationsFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.DetailedJobOutIntegrations$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DetailedJobOutIntegrations' from JSON`);
}
/** @internal */
exports.DetailedJobOutRepositories$inboundSchema = githubrepositoryout_js_1.GithubRepositoryOut$inboundSchema;
/** @internal */
exports.DetailedJobOutRepositories$outboundSchema = githubrepositoryout_js_1.GithubRepositoryOut$outboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var DetailedJobOutRepositories$;
(function (DetailedJobOutRepositories$) {
    /** @deprecated use `DetailedJobOutRepositories$inboundSchema` instead. */
    DetailedJobOutRepositories$.inboundSchema = exports.DetailedJobOutRepositories$inboundSchema;
    /** @deprecated use `DetailedJobOutRepositories$outboundSchema` instead. */
    DetailedJobOutRepositories$.outboundSchema = exports.DetailedJobOutRepositories$outboundSchema;
})(DetailedJobOutRepositories$ || (exports.DetailedJobOutRepositories$ = DetailedJobOutRepositories$ = {}));
function detailedJobOutRepositoriesToJSON(detailedJobOutRepositories) {
    return JSON.stringify(exports.DetailedJobOutRepositories$outboundSchema.parse(detailedJobOutRepositories));
}
function detailedJobOutRepositoriesFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.DetailedJobOutRepositories$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DetailedJobOutRepositories' from JSON`);
}
/** @internal */
exports.DetailedJobOut$inboundSchema = z.object({
    id: z.string(),
    auto_start: z.boolean(),
    hyperparameters: trainingparameters_js_1.TrainingParameters$inboundSchema,
    model: z.string(),
    status: exports.DetailedJobOutStatus$inboundSchema,
    job_type: z.string(),
    created_at: z.number().int(),
    modified_at: z.number().int(),
    training_files: z.array(z.string()),
    validation_files: z.nullable(z.array(z.string())).optional(),
    object: z.literal("job").default("job"),
    fine_tuned_model: z.nullable(z.string()).optional(),
    suffix: z.nullable(z.string()).optional(),
    integrations: z.nullable(z.array(wandbintegrationout_js_1.WandbIntegrationOut$inboundSchema))
        .optional(),
    trained_tokens: z.nullable(z.number().int()).optional(),
    repositories: z.array(githubrepositoryout_js_1.GithubRepositoryOut$inboundSchema).optional(),
    metadata: z.nullable(jobmetadataout_js_1.JobMetadataOut$inboundSchema).optional(),
    events: z.array(eventout_js_1.EventOut$inboundSchema).optional(),
    checkpoints: z.array(checkpointout_js_1.CheckpointOut$inboundSchema).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "auto_start": "autoStart",
        "job_type": "jobType",
        "created_at": "createdAt",
        "modified_at": "modifiedAt",
        "training_files": "trainingFiles",
        "validation_files": "validationFiles",
        "fine_tuned_model": "fineTunedModel",
        "trained_tokens": "trainedTokens",
    });
});
/** @internal */
exports.DetailedJobOut$outboundSchema = z.object({
    id: z.string(),
    autoStart: z.boolean(),
    hyperparameters: trainingparameters_js_1.TrainingParameters$outboundSchema,
    model: z.string(),
    status: exports.DetailedJobOutStatus$outboundSchema,
    jobType: z.string(),
    createdAt: z.number().int(),
    modifiedAt: z.number().int(),
    trainingFiles: z.array(z.string()),
    validationFiles: z.nullable(z.array(z.string())).optional(),
    object: z.literal("job").default("job"),
    fineTunedModel: z.nullable(z.string()).optional(),
    suffix: z.nullable(z.string()).optional(),
    integrations: z.nullable(z.array(wandbintegrationout_js_1.WandbIntegrationOut$outboundSchema))
        .optional(),
    trainedTokens: z.nullable(z.number().int()).optional(),
    repositories: z.array(githubrepositoryout_js_1.GithubRepositoryOut$outboundSchema).optional(),
    metadata: z.nullable(jobmetadataout_js_1.JobMetadataOut$outboundSchema).optional(),
    events: z.array(eventout_js_1.EventOut$outboundSchema).optional(),
    checkpoints: z.array(checkpointout_js_1.CheckpointOut$outboundSchema).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        autoStart: "auto_start",
        jobType: "job_type",
        createdAt: "created_at",
        modifiedAt: "modified_at",
        trainingFiles: "training_files",
        validationFiles: "validation_files",
        fineTunedModel: "fine_tuned_model",
        trainedTokens: "trained_tokens",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var DetailedJobOut$;
(function (DetailedJobOut$) {
    /** @deprecated use `DetailedJobOut$inboundSchema` instead. */
    DetailedJobOut$.inboundSchema = exports.DetailedJobOut$inboundSchema;
    /** @deprecated use `DetailedJobOut$outboundSchema` instead. */
    DetailedJobOut$.outboundSchema = exports.DetailedJobOut$outboundSchema;
})(DetailedJobOut$ || (exports.DetailedJobOut$ = DetailedJobOut$ = {}));
function detailedJobOutToJSON(detailedJobOut) {
    return JSON.stringify(exports.DetailedJobOut$outboundSchema.parse(detailedJobOut));
}
function detailedJobOutFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.DetailedJobOut$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'DetailedJobOut' from JSON`);
}
