// @mistralai/mistralai/models/components/function.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.FunctionT$ = exports.FunctionT$outboundSchema = exports.FunctionT$inboundSchema = void 0;
exports.functionToJSON = functionToJSON;
exports.functionFromJSON = functionFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.FunctionT$inboundSchema = z.object({
    name: z.string(),
    description: z.string().default(""),
    strict: z.boolean().default(false),
    parameters: z.record(z.any()),
});
/** @internal */
exports.FunctionT$outboundSchema = z.object({
    name: z.string(),
    description: z.string().default(""),
    strict: z.boolean().default(false),
    parameters: z.record(z.any()),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var FunctionT$;
(function (FunctionT$) {
    /** @deprecated use `FunctionT$inboundSchema` instead. */
    FunctionT$.inboundSchema = exports.FunctionT$inboundSchema;
    /** @deprecated use `FunctionT$outboundSchema` instead. */
    FunctionT$.outboundSchema = exports.FunctionT$outboundSchema;
})(FunctionT$ || (exports.FunctionT$ = FunctionT$ = {}));
function functionToJSON(functionT) {
    return JSON.stringify(exports.FunctionT$outboundSchema.parse(functionT));
}
function functionFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.FunctionT$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'FunctionT' from JSON`);
}
