// @mistralai/mistralai/models/components/jobsout.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.JobsOut$ = exports.JobsOut$outboundSchema = exports.JobsOut$inboundSchema = exports.JobsOutObject$ = exports.JobsOutObject$outboundSchema = exports.JobsOutObject$inboundSchema = exports.JobsOutObject = void 0;
exports.jobsOutToJSON = jobsOutToJSON;
exports.jobsOutFromJSON = jobsOutFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
const jobout_js_1 = require("./jobout.js");
exports.JobsOutObject = {
    List: "list",
};
/** @internal */
exports.JobsOutObject$inboundSchema = z.nativeEnum(exports.JobsOutObject);
/** @internal */
exports.JobsOutObject$outboundSchema = exports.JobsOutObject$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var JobsOutObject$;
(function (JobsOutObject$) {
    /** @deprecated use `JobsOutObject$inboundSchema` instead. */
    JobsOutObject$.inboundSchema = exports.JobsOutObject$inboundSchema;
    /** @deprecated use `JobsOutObject$outboundSchema` instead. */
    JobsOutObject$.outboundSchema = exports.JobsOutObject$outboundSchema;
})(JobsOutObject$ || (exports.JobsOutObject$ = JobsOutObject$ = {}));
/** @internal */
exports.JobsOut$inboundSchema = z.object({
    data: z.array(jobout_js_1.JobOut$inboundSchema).optional(),
    object: z.literal("list").default("list"),
    total: z.number().int(),
});
/** @internal */
exports.JobsOut$outboundSchema = z.object({
    data: z.array(jobout_js_1.JobOut$outboundSchema).optional(),
    object: z.literal("list").default("list"),
    total: z.number().int(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var JobsOut$;
(function (JobsOut$) {
    /** @deprecated use `JobsOut$inboundSchema` instead. */
    JobsOut$.inboundSchema = exports.JobsOut$inboundSchema;
    /** @deprecated use `JobsOut$outboundSchema` instead. */
    JobsOut$.outboundSchema = exports.JobsOut$outboundSchema;
})(JobsOut$ || (exports.JobsOut$ = JobsOut$ = {}));
function jobsOutToJSON(jobsOut) {
    return JSON.stringify(exports.JobsOut$outboundSchema.parse(jobsOut));
}
function jobsOutFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.JobsOut$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'JobsOut' from JSON`);
}
