// @mistralai/mistralai/models/operations/filesapiroutesretrievefile.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.FilesApiRoutesRetrieveFileRequest$ = exports.FilesApiRoutesRetrieveFileRequest$outboundSchema = exports.FilesApiRoutesRetrieveFileRequest$inboundSchema = void 0;
exports.filesApiRoutesRetrieveFileRequestToJSON = filesApiRoutesRetrieveFileRequestToJSON;
exports.filesApiRoutesRetrieveFileRequestFromJSON = filesApiRoutesRetrieveFileRequestFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.FilesApiRoutesRetrieveFileRequest$inboundSchema = z.object({
    file_id: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "file_id": "fileId",
    });
});
/** @internal */
exports.FilesApiRoutesRetrieveFileRequest$outboundSchema = z.object({
    fileId: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        fileId: "file_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var FilesApiRoutesRetrieveFileRequest$;
(function (FilesApiRoutesRetrieveFileRequest$) {
    /** @deprecated use `FilesApiRoutesRetrieveFileRequest$inboundSchema` instead. */
    FilesApiRoutesRetrieveFileRequest$.inboundSchema = exports.FilesApiRoutesRetrieveFileRequest$inboundSchema;
    /** @deprecated use `FilesApiRoutesRetrieveFileRequest$outboundSchema` instead. */
    FilesApiRoutesRetrieveFileRequest$.outboundSchema = exports.FilesApiRoutesRetrieveFileRequest$outboundSchema;
})(FilesApiRoutesRetrieveFileRequest$ || (exports.FilesApiRoutesRetrieveFileRequest$ = FilesApiRoutesRetrieveFileRequest$ = {}));
function filesApiRoutesRetrieveFileRequestToJSON(filesApiRoutesRetrieveFileRequest) {
    return JSON.stringify(exports.FilesApiRoutesRetrieveFileRequest$outboundSchema.parse(filesApiRoutesRetrieveFileRequest));
}
function filesApiRoutesRetrieveFileRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.FilesApiRoutesRetrieveFileRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'FilesApiRoutesRetrieveFileRequest' from JSON`);
}
