// @mistralai/mistralai/models/operations/filesapiroutesdeletefile.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.FilesApiRoutesDeleteFileRequest$ = exports.FilesApiRoutesDeleteFileRequest$outboundSchema = exports.FilesApiRoutesDeleteFileRequest$inboundSchema = void 0;
exports.filesApiRoutesDeleteFileRequestToJSON = filesApiRoutesDeleteFileRequestToJSON;
exports.filesApiRoutesDeleteFileRequestFromJSON = filesApiRoutesDeleteFileRequestFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.FilesApiRoutesDeleteFileRequest$inboundSchema = z.object({
    file_id: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "file_id": "fileId",
    });
});
/** @internal */
exports.FilesApiRoutesDeleteFileRequest$outboundSchema = z.object({
    fileId: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        fileId: "file_id",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var FilesApiRoutesDeleteFileRequest$;
(function (FilesApiRoutesDeleteFileRequest$) {
    /** @deprecated use `FilesApiRoutesDeleteFileRequest$inboundSchema` instead. */
    FilesApiRoutesDeleteFileRequest$.inboundSchema = exports.FilesApiRoutesDeleteFileRequest$inboundSchema;
    /** @deprecated use `FilesApiRoutesDeleteFileRequest$outboundSchema` instead. */
    FilesApiRoutesDeleteFileRequest$.outboundSchema = exports.FilesApiRoutesDeleteFileRequest$outboundSchema;
})(FilesApiRoutesDeleteFileRequest$ || (exports.FilesApiRoutesDeleteFileRequest$ = FilesApiRoutesDeleteFileRequest$ = {}));
function filesApiRoutesDeleteFileRequestToJSON(filesApiRoutesDeleteFileRequest) {
    return JSON.stringify(exports.FilesApiRoutesDeleteFileRequest$outboundSchema.parse(filesApiRoutesDeleteFileRequest));
}
function filesApiRoutesDeleteFileRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.FilesApiRoutesDeleteFileRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'FilesApiRoutesDeleteFileRequest' from JSON`);
}
