// @mistralai/mistralai/models/operations/jobsapiroutesbatchgetbatchjobs.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.JobsApiRoutesBatchGetBatchJobsRequest$ = exports.JobsApiRoutesBatchGetBatchJobsRequest$outboundSchema = exports.JobsApiRoutesBatchGetBatchJobsRequest$inboundSchema = void 0;
exports.jobsApiRoutesBatchGetBatchJobsRequestToJSON = jobsApiRoutesBatchGetBatchJobsRequestToJSON;
exports.jobsApiRoutesBatchGetBatchJobsRequestFromJSON = jobsApiRoutesBatchGetBatchJobsRequestFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
const components = __importStar(require("../components/index.js"));
/** @internal */
exports.JobsApiRoutesBatchGetBatchJobsRequest$inboundSchema = z.object({
    page: z.number().int().default(0),
    page_size: z.number().int().default(100),
    model: z.nullable(z.string()).optional(),
    metadata: z.nullable(z.record(z.any())).optional(),
    created_after: z.nullable(z.string().datetime({ offset: true }).transform(v => new Date(v))).optional(),
    created_by_me: z.boolean().default(false),
    status: z.nullable(components.BatchJobStatus$inboundSchema).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "page_size": "pageSize",
        "created_after": "createdAfter",
        "created_by_me": "createdByMe",
    });
});
/** @internal */
exports.JobsApiRoutesBatchGetBatchJobsRequest$outboundSchema = z.object({
    page: z.number().int().default(0),
    pageSize: z.number().int().default(100),
    model: z.nullable(z.string()).optional(),
    metadata: z.nullable(z.record(z.any())).optional(),
    createdAfter: z.nullable(z.date().transform(v => v.toISOString())).optional(),
    createdByMe: z.boolean().default(false),
    status: z.nullable(components.BatchJobStatus$outboundSchema).optional(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        pageSize: "page_size",
        createdAfter: "created_after",
        createdByMe: "created_by_me",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var JobsApiRoutesBatchGetBatchJobsRequest$;
(function (JobsApiRoutesBatchGetBatchJobsRequest$) {
    /** @deprecated use `JobsApiRoutesBatchGetBatchJobsRequest$inboundSchema` instead. */
    JobsApiRoutesBatchGetBatchJobsRequest$.inboundSchema = exports.JobsApiRoutesBatchGetBatchJobsRequest$inboundSchema;
    /** @deprecated use `JobsApiRoutesBatchGetBatchJobsRequest$outboundSchema` instead. */
    JobsApiRoutesBatchGetBatchJobsRequest$.outboundSchema = exports.JobsApiRoutesBatchGetBatchJobsRequest$outboundSchema;
})(JobsApiRoutesBatchGetBatchJobsRequest$ || (exports.JobsApiRoutesBatchGetBatchJobsRequest$ = JobsApiRoutesBatchGetBatchJobsRequest$ = {}));
function jobsApiRoutesBatchGetBatchJobsRequestToJSON(jobsApiRoutesBatchGetBatchJobsRequest) {
    return JSON.stringify(exports.JobsApiRoutesBatchGetBatchJobsRequest$outboundSchema.parse(jobsApiRoutesBatchGetBatchJobsRequest));
}
function jobsApiRoutesBatchGetBatchJobsRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.JobsApiRoutesBatchGetBatchJobsRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'JobsApiRoutesBatchGetBatchJobsRequest' from JSON`);
}
