// @raycast/eslint-plugin/dist/title-case.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.titleCase = void 0;
function titleCase(s) {
    const noCaps = {
        and: true,
        but: true,
        or: true,
        nor: true,
        for: true,
        yet: true,
        so: true,
        to: true,
        as: true,
        at: true,
        by: true,
        from: true,
        in: true,
        into: true,
        of: true,
        off: true,
        on: true,
        onto: true,
        out: true,
        over: true,
        up: true,
        with: true,
    };
    const articles = {
        a: true,
        an: true,
        the: true,
    };
    const fixedCaseWords = {
        npm: true,
        "crates.io": true,
        dbt: true,
        "pub.dev": true,
        kubectx: true,
        "monday.com": true,
        "ray.so": true,
        flomo: true,
        iterm: true,
        xkcd: true,
        macos: true,
        iphone: true,
        github: true,
        ide: true,
        url: true,
        vs: true,
        ai: true,
    };
    const fixedCaseWordsMap = {
        npm: "npm",
        "crates.io": "crates.io",
        dbt: "dbt",
        "pub.dev": "pub.dev",
        kubectx: "kubectx",
        "monday.com": "monday.com",
        "ray.so": "ray.so",
        flomo: "flomo",
        iterm: "iTerm",
        xkcd: "xkcd",
        macos: "macOS",
        iphone: "iPhone",
        github: "GitHub",
        ide: "IDE",
        url: "URL",
        vs: "VS",
        ai: "AI",
    };
    s = s.replace(/\.\.\./g, "…");
    const words = s.split(" ");
    for (let i = 0; i < words.length; i++) {
        const word = words[i];
        const lowerWord = word.toLowerCase();
        const ok = noCaps[lowerWord];
        const isArticle = articles[lowerWord];
        const isFixedCase = fixedCaseWords[lowerWord];
        if (isFixedCase) {
            words[i] = fixedCaseWordsMap[lowerWord];
        }
        else if ((!ok && !isArticle) ||
            i === 0 ||
            i === words.length - 1 ||
            (isArticle && words[i - 1].endsWith(":"))) {
            words[i] = word.charAt(0).toUpperCase() + word.slice(1).toLowerCase();
        }
        else {
            words[i] = word.toLowerCase();
        }
    }
    return words.join(" ");
}
exports.titleCase = titleCase;
